code/modules/reagents/Chemistry-Reagents/Drug-Reagents.dm

```diff
@@ -162,46 +162,33 @@
 /datum/reagent/drug/methamphetamine
 	name = "Methamphetamine"
 	id = "methamphetamine"
-	description = "Reduces stun times by about 300%, speeds the user up, and allows the user to quickly recover stamina while dealing a small amount of Brain damage. If overdosed the subject will move randomly, laugh randomly, drop items and suffer from Toxin and Brain damage. If addicted the subject will constantly jitter and drool, before becoming dizzy and losing motor control and eventually suffer heavy toxin damage."
+	description = "Halves the duration of stuns and greatly increases movement speed. However, it drains an ever-increasing amount of stamina over time and being stunned will force the body to metabolize the drug faster."
 	reagent_state = LIQUID
 	color = "#60A584" // rgb: 96, 165, 132
-	overdose_threshold = 15
-	addiction_threshold = 10
+	overdose_threshold = 10
 	metabolization_rate = 0.75 * REAGENTS_METABOLISM
 
 /datum/reagent/drug/methamphetamine/on_mob_life(mob/living/M)
 	var/high_message = pick("You feel hyper.", "You feel like you need to go faster.", "You feel like you can run the world.")
-	var/tolerance_message = pick("You don't feel so good.", "You feel a bit sick.", "You feel like you've had better days.")
-	if(current_cycle <= 50)
-		if(prob(5))
-			M << "<span class='notice'>[high_message]</span>"
-		M.AdjustParalysis(-2)
-		M.AdjustStunned(-2)
-		M.AdjustWeakened(-2)
-		M.adjustStaminaLoss(-2)
-		M.status_flags |= GOTTAGOREALLYFAST
-	if(current_cycle > 50)
-		if(prob(15))
-			M << "<span class='notice'>[tolerance_message]</span>"
-		M.adjustStaminaLoss(3)
-		M.adjustToxLoss(1*REM)
-	M.Jitter(2)
+	if(prob(5))
+		M << "<span class='notice'>[high_message]</span>"
+	M.AdjustParalysis(-1)
+	M.AdjustStunned(-1)
+	M.AdjustWeakened(-1)
+	M.adjustStaminaLoss(4) // Is actually -2 per tick, humans regenerate 2 per tick
+	M.adjustToxLoss(0.4*REM)
+	M.status_flags |= GOTTAGOREALLYFAST
+	M.sleeping = max(0,M.sleeping - 3)
 	if(prob(5))
 		M.emote(pick("twitch", "shiver"))
+	M.Jitter(2)
+	if(M.stunned || M.weakened)
+		holder.remove_reagent("methamphetamine", 0.75 * REAGENTS_METABOLISM)
```

Comment:
Why not use metabolization_rate as defined for this drug? It means you can also remove the snowflakey code below for heroin (Set its metabolization_rate to 2 * REAGENTS_METABOLISM)